src/Database/Type.php

```diff
@@ -21,7 +21,7 @@
  * Encapsulates all conversion functions for values coming from database into PHP and
  * going from PHP into database.
  */
-class Type
+class Type implements TypeInterface
```

Comment:
Can we move this line to each of the subclasses? Maybe this is a good chance to start closing this class for inheritance and only use it as the type builder